pallets/phala/src/registry.rs

```diff
@@ -373,6 +381,47 @@ pub mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(0)]
+		pub fn update_worker_endpoint(
+			origin: OriginFor<T>,
+			pubkey: WorkerPublicKey,
+			versioned_endpoints: VersionedWorkerEndpoint,
+			signature: Vec<u8>,
```

Comment:
I suggest to wrap them as a struct like this:

```rust
struct SignedEndpoint {
  pubkey: WorkerPublicKey,
  endpoints: VersionedWorkerEndpoint,
  signing_time: BlockNumber,
  signature: Vec<u8>,
}
```

And then, you can remove the block number in `VersionedWorkerEndpoint` as it's only used in signature verification.